internal/server/handler_v2.go

```diff
@@ -49,26 +49,51 @@ func (s *Server) V2Node(
 		if !arc.Out {
 			direction = util.DirectionIn
 		}
-		if arc.SingleProp == "" && len(arc.BracketProps) == 0 {
-			// Examples:
-			//   ->
-			//   <-
-			return v2p.API(ctx, s.store, in.GetNodes(), direction)
-		}
-		if arc.SingleProp != "" && arc.Wildcard == "" {
+		if arc.SingleProp != "" {
+			if arc.Wildcard == "" {
+				// Examples:
+				//   ->name
+				//   <-containedInPlace
+				return v2pv.API(
+					ctx,
+					s.store,
+					in.GetNodes(),
+					[]string{arc.SingleProp},
+					direction,
+					int(in.GetLimit()),
+					in.NextToken,
+				)
+			}
+
 			// Examples:
-			//   ->name
-			//   <-containedInPlace
-			return v2pv.API(
-				ctx,
-				s.store,
-				in.GetNodes(),
-				[]string{arc.SingleProp},
-				direction,
-				int(in.GetLimit()),
-				in.NextToken,
-			)
-		} else if len(arc.BracketProps) > 0 {
+			//   <-containedInPlace+{typeOf:City}
+			if arc.Wildcard != "" && arc.Wildcard != "+" {
+				return nil, status.Errorf(codes.InvalidArgument,
+					"only '+' wildcard is supported")
+			}
+			if arc.SingleProp != "containedInPlace" {
+				return nil, status.Errorf(codes.InvalidArgument,
+					"only containedInPlace is supported for wildcard '+'")
+			}
+			if arc.Out {
+				return nil, status.Errorf(codes.InvalidArgument,
+					"only in arc is supported for containedInPlace+")
+			}
+			placeType, ok := arc.Filter["typeOf"]
```

Comment:
shall we put all the working scenarios at the beginning, so it's clear to see all the exact working scenarios. Then return errors for the rest of the case?